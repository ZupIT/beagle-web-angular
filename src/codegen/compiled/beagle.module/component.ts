/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Type } from '@angular/core'
import {
  kebabToCamelCase,
  removeExtraIndentation,
  replaceToUnderline,
} from '../../utils/formatting'
import { getComponentAnnotations } from '../../utils/metadata'
import { remoteViewSelector } from '../../../constants'


function createViewChildString(name: string, templateName: string, angularVersion: number): string {
  const componentName = replaceToUnderline(name)

  if (angularVersion >= 8) {
    return `'${componentName}': new ViewChild('${templateName}', { static: true })`
  } else {
    return `'${componentName}': new ViewChild('${templateName}')`
  }
}

function createQueries(components: Record<string, Type<any>>, angularVersion: number) {
  const componentNames = Object.keys(components)

  return componentNames.map((name) => {
    const selector = getComponentAnnotations(components[name]).selector
    const templateName = kebabToCamelCase(selector)
    return createViewChildString(name, templateName, angularVersion)
  })
}


export function createComponentString(
    components: Record<string, Type<any>>,
    angularVersion: number,
  ) {
  const queries = createQueries(components, angularVersion)

  // todo: legacy code. Remove the input "loadParams" with v2.0.
  const componentString = `
    @Component({
      selector: '${remoteViewSelector}',
      template,
      inputs: ['loadParams', 'route', 'networkOptions', 'controllerId'],
      queries: {
        ${queries.join(',\n        ')},
      },
    })
    export class BeagleRemoteView extends AbstractBeagleRemoteView {
      constructor(
        beagleProvider: BeagleProvider,
        ngZone: NgZone,
        changeDetector: ChangeDetectorRef,
      ) {
        // @ts-ignore
        super(beagleProvider, ngZone, changeDetector)
      }
    }
  `

  return removeExtraIndentation(componentString, 4)
}
