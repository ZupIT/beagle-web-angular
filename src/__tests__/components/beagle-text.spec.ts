/*
  * Copyright 2020, 2022 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { TestBed, async, ComponentFixture } from '@angular/core/testing'
import { BeagleTextComponent } from '../../components/beagle-text/beagle-text.component'
import { buildBeagleTestModuleMetadata } from './test-module-builder'

describe('BeagleTextComponent', () => {
  let component: BeagleTextComponent
  let fixture: ComponentFixture<BeagleTextComponent>

  const initialText = 'Test'

  beforeEach(async(() => {
    TestBed.configureTestingModule(buildBeagleTestModuleMetadata([BeagleTextComponent]))
      .compileComponents()
  }))

  beforeEach(() => {
    fixture = TestBed.createComponent(BeagleTextComponent)
    component = fixture.componentInstance    
  })

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('should render the string as initialized', () => {
    component.text = initialText
    fixture.detectChanges()

    const p = fixture.debugElement.nativeElement.querySelector('p')
    expect(p?.innerHTML).toBe(initialText)
  })

  it('should not render any text when text is null', () => {
    const p = fixture.debugElement.nativeElement.querySelector('p')

    component.text = null
    fixture.detectChanges()
    expect(p?.innerHTML.length).toBe(0)
  })

  it('should not render any text when text is undefined', () => {
    const p = fixture.debugElement.nativeElement.querySelector('p')

    component.text = undefined
    fixture.detectChanges()
    expect(p?.innerHTML.length).toBe(0)
  })

  it('should render numbers as strings', () => {
    const p = fixture.debugElement.nativeElement.querySelector('p')
    const value = 123.45

    component.text = value
    fixture.detectChanges()
    expect(p?.innerHTML).toBe(String(value))
  })

  it('should render an object as stringfied json object', () => {
    const p = fixture.debugElement.nativeElement.querySelector('p')
    const obj = { my: 'test' }

    component.text = obj
    fixture.detectChanges()
    expect(p?.innerHTML).toBe(JSON.stringify(obj))
  })

  it('should render an array as stringfied json array', () => {
    const p = fixture.debugElement.nativeElement.querySelector('p')
    const arr = [{ my: 'test' }, 'test', 123.45]

    component.text = arr
    fixture.detectChanges()
    expect(p?.innerHTML).toBe(JSON.stringify(arr))
  })

  it('should render a function as empty string', () => {
    const p = fixture.debugElement.nativeElement.querySelector('p')

    component.text = function test() { return true }
    fixture.detectChanges()
    expect(p?.innerHTML).toBe('')
  })
})
