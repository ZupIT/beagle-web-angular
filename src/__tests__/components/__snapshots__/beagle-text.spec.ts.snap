// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeagleTextComponent should match snapshot 1`] = `
BeagleTextComponent {
  "alignment": "INHERIT",
  "styleId": "",
  "text": "Teste",
  "textColor": "inherit",
}
`;
